src/rgw/rgw_data_sync.cc

```diff
@@ -672,6 +706,34 @@ int RGWRemoteDataLog::read_sync_status(rgw_data_sync_status *sync_status)
   return ret;
 }
 
+int RGWRemoteDataLog::read_recovering_shards(rgw_data_sync_status *sync_status, set<int>& recovering_shards)
```

Comment:
it looks like this function only uses `sync_status->sync_info.num_shards` from `sync_status`, could you just pass `int num_shards` instead?